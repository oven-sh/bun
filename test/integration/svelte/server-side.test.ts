// TODO: full server-side support
// import { SveltePlugin } from "bun-plugin-svelte";
// import { render } from "svelte/server";
// import { bunRun, bunEnv, bunExe } from "harness";
// import path from "path";
// // import { describe, beforeEach, afterEach, it, expect } from "bun:test";

// const fixturePath = (...segs: string[]) => path.join(__dirname, "fixtures", ...segs);

// // await Bun.plugin(SveltePlugin({ development: true }));

// // import TodoApp from "./fixtures/todo-list.svelte";

// // afterAll(() => {
// //   Bun.plugin.clearAll();
// // })

// describe("When bun-plugin-svelte is enabled via Bun.plugin()", () => {
//   // beforeEach(async () => {
//   //   await Bun.plugin(SveltePlugin({ development: true }));
//   // });

//   // afterEach(() => {
//   //   Bun.plugin.clearAll();
//   // });

//   it("can render() production builds", async () => {
//     const result = Bun.spawnSync([bunExe(), "--preload=./server-imports.preload.ts", "server-imports.ts"], {
//       cwd: fixturePath(),
//       env: bunEnv,
//     });
//     if (result.exitCode !== 0) {
//       console.error(result.stderr.toString("utf8"));
//       throw new Error("rendering failed");
//     }
//     expect(result.exitCode).toBe(0);

//     // const { default: TodoApp } = await import("./fixtures/todo-list.svelte");
//     // expect(TodoApp).toBeTypeOf("function");
//     // const result = render(TodoApp);
//     // expect(result).toMatchObject({ head: expect.any(String), body: expect.any(String) });
//     // expect(result).toMatchSnapshot();
//   });

//   it("can render() development builds", async () => {
//     const result = Bun.spawnSync([bunExe(), "--preload=./server-imports.preload.ts", "server-imports.ts"], {
//       cwd: fixturePath(),
//       env: {
//         ...bunEnv,
//         NODE_ENV: "development",
//       }
//     });
//     if (result.exitCode !== 0) {
//       console.error(result.stderr.toString("utf8"));
//       throw new Error("rendering failed");
//     }
//     expect(result.exitCode).toBe(0);

//     // // const { default: TodoApp } = await import("./fixtures/todo-list.svelte");
//     // const result = render(TodoApp);
//     // expect(result).toMatchObject({ head: expect.any(String), body: expect.any(String) });
//     // expect(result).toMatchSnapshot();
//   });

//   // FIXME: onResolve is not called for CSS imports on server-side
//   it.skip("if forced to use client-side generation, could be used with happy-dom in Bun", () => {
//     expect(() => bunRun(fixturePath("client-code-on-server.ts"), { NODE_ENV: "development" })).not.toThrow();
//   })
// });

// // describe("When using Bun.build()", () => {


// // });
