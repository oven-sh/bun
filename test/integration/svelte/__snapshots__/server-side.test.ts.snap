// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`When bun-plugin-svelte is enabled via Bun.plugin() can be render()-ed 1`] = `
{
  "body": Any<String>,
  "head": Any<String>,
  "html": 
"<!--[--><!-- source: https://svelte.dev/playground/7eb8c1dd6cac414792b0edb53521ab49?version=5.20.4 -->


<main>
  <h1 class="title svelte-y13uzakzmn8w">Todo List</h1>
  <input value="" type="text" placeholder="new todo item..">
  <button>Add</button>

  <br>
  <!--[--><!---->
    <input checked type="checkbox">
    <span class="svelte-y13uzakzmn8w checked">Write my first post</span>
    <span role="button">❌</span>
    <br>
  <!---->
    <input type="checkbox">
    <span class="svelte-y13uzakzmn8w">Upload the post to the blog</span>
    <span role="button">❌</span>
    <br>
  <!---->
    <input type="checkbox">
    <span class="svelte-y13uzakzmn8w">Publish the post at Facebook</span>
    <span role="button">❌</span>
    <br>
  <!--]-->
</main>

<!--]-->"
,
}
`;
